irispy/sji.py

```diff
@@ -878,3 +879,69 @@ def dustbuster(mc):
         map.data = image_orig
 
     return mc
+
+
+def read_iris_sji_level2_fits(filename):
+    """
+    Read IRIS level 2 SJI FITS from an OBS into an NDCube instance
+
+    Parameters
+    ----------
+    filename : `str`
+        File name to be read
+
+    Returns
+    -------
+    result: 'ndcube.NDCube'
+
+    """
+
+    # Open a fits file
+    my_file = fits.open(filename)
+    # Derive WCS, data and mask for NDCube from fits file.
+    wcs = WCS(my_file[0].header)
+    data = my_file[0].data
+    data_nan_masked = my_file[0].data
+    data_nan_masked[data == BAD_PIXEL_VALUE] = np.nan
+    mask = data_nan_masked == BAD_PIXEL_VALUE
+    # Derive unit and readout noise from detector.
+    exposure_times = my_file[1].data[:, my_file[1].header["EXPTIMES"]]
+    unit = iris_tools.DN_UNIT["SJI"]
+    readout_noise = iris_tools.READOUT_NOISE["SJI"]
+    # Derive uncertainty of data for NDCube from fits file.
+    uncertainty = u.Quantity(np.sqrt((data_nan_masked*unit).to(u.photon).value
+                                     + readout_noise.to(u.photon).value**2),
+                             unit=u.photon).to(unit).value
+    # Derive extra coordinates for NDCube from fits file.
+    times = np.array([parse_time(my_file[0].header["STARTOBS"])
+                      + timedelta(seconds=s)
+                      for s in my_file[1].data[:, my_file[1].header["TIME"]]])
+    pztx = my_file[1].data[:, my_file[1].header["PZTX"]] * u.arcsec
+    pzty = my_file[1].data[:, my_file[1].header["PZTY"]] * u.arcsec
+    xcenix = my_file[1].data[:, my_file[1].header["XCENIX"]] * u.arcsec
+    ycenix = my_file[1].data[:, my_file[1].header["YCENIX"]] * u.arcsec
+    obs_vrix = my_file[1].data[:, my_file[1].header["OBS_VRIX"]] * u.m/u.s
+    ophaseix = my_file[1].data[:, my_file[1].header["OPHASEIX"]]
+    extra_coords = [('TIME', 0, times), ("PZTX", 0, pztx), ("PZTY", 0, pzty),
+                    ("XCENIX", 0, xcenix), ("YCENIX", 0, ycenix),
+                    ("OBS_VRIX", 0, obs_vrix), ("OPHASEIX", 0, ophaseix),
+                    ("EXPOSURE TIME", 0, exposure_times)]
+    # Extraction of meta for NDCube from fits file.
+    date_obs = my_file[0].header.get('DATE_OBS', None)
+    date_obs = parse_time(date_obs) if date_obs else None
+    date_end = my_file[0].header.get('DATE_END', None)
+    date_end = parse_time(date_end) if date_end else None
+    meta = {'TELESCOP': my_file[0].header.get('TELESCOP'),
+            'INSTRUME': my_file[0].header.get('INSTRUME'),
+            'TWAVE1': my_file[0].header.get('TWAVE1'),
+            'WAVEUNIT': "Angstrom",
```

Comment:
```SJICube``` (which the output of this function will become) should not contain any spectrograph-specific metadata.  So I think it's best to remove ```'WAVEUNIT'```.  Same goes for any other spectrograph-specific metadata.

Reply:
Ok, I will remove it.